tests/test_integration/test_run_transfer/test_mssql.py

```diff
@@ -0,0 +1,296 @@
+        session=session,
+        group_id=group.id,
+        name=f"postgres2mssql_{secrets.token_hex(5)}",
+        source_connection_id=postgres_connection.id,
+        target_connection_id=mssql_connection.id,
+        source_params={
+            "type": "postgres",
+            "table_name": "public.source_table",
+        },
+        target_params={
+            "type": "mssql",
+            "table_name": "dbo.target_table",
+        },
+        queue_id=queue.id,
+    )
+    yield result
+    await session.delete(result)
+    await session.commit()
+
+
+@pytest_asyncio.fixture
+async def mssql_to_postgres(
+    session: AsyncSession,
+    group: Group,
+    queue: Queue,
+    mssql_for_conftest: MSSQL,
+    mssql_connection: Connection,
+    postgres_connection: Connection,
+):
+    result = await create_transfer(
+        session=session,
+        group_id=group.id,
+        name=f"mssql2postgres_{secrets.token_hex(5)}",
+        source_connection_id=mssql_connection.id,
+        target_connection_id=postgres_connection.id,
+        source_params={
+            "type": "mssql",
+            "table_name": "dbo.source_table",
+        },
+        target_params={
+            "type": "postgres",
+            "table_name": "public.target_table",
+        },
+        queue_id=queue.id,
+    )
+    yield result
+    await session.delete(result)
+    await session.commit()
+
+
+async def test_run_transfer_postgres_to_mssql(
+    client: AsyncClient,
+    group_owner: MockUser,
+    prepare_postgres,
+    prepare_mssql,
+    init_df: DataFrame,
+    postgres_to_mssql: Transfer,
+):
+    # Arrange
+    _, fill_with_data = prepare_postgres
+    fill_with_data(init_df)
+    mssql, _ = prepare_mssql
+
+    # Act
+    result = await client.post(
+        "v1/runs",
+        headers={"Authorization": f"Bearer {group_owner.token}"},
+        json={"transfer_id": postgres_to_mssql.id},
+    )
+    # Assert
+    assert result.status_code == 200
+
+    run_data = await get_run_on_end(
+        client=client,
+        run_id=result.json()["id"],
+        token=group_owner.token,
+    )
+    source_auth_data = run_data["transfer_dump"]["source_connection"]["auth_data"]
+    target_auth_data = run_data["transfer_dump"]["target_connection"]["auth_data"]
+
+    assert run_data["status"] == Status.FINISHED.value
+    assert source_auth_data["user"]
+    assert "password" not in source_auth_data
+    assert target_auth_data["user"]
+    assert "password" not in target_auth_data
+    reader = DBReader(
+        connection=mssql,
+        table="dbo.target_table",
+    )
+    df = reader.run()
+
+    # as spark rounds datetime up to milliseconds while writing to mssql: https://onetl.readthedocs.io/en/latest/connection/db_connection/mssql/types.html#id5
+    df = df.withColumn("REGISTERED_AT", date_trunc("second", col("REGISTERED_AT")))
+    init_df = init_df.withColumn("REGISTERED_AT", date_trunc("second", col("REGISTERED_AT")))
```

Comment:
Is that really needed?

Reply:
In MSSQL the datetime type [rounds values](https://learn.microsoft.com/en-us/sql/t-sql/data-types/datetime-transact-sql?view=sql-server-ver16#:~:text=8%20bytes-,Accuracy,-Rounded%20to%20increments) to the nearest increment of 0.000, 0.003, or 0.007 seconds (approx. 3.33 milliseconds) due to its precision limitation. For example:

- `id=7282, 2006-02-14 12:36:15.589307` is rounded to 590 milliseconds
- `id=815, 1998-11-10 01:46:12.228133` is rounded to 227 milliseconds

Therefore I decided to drop milliseconds in tests

Reply:
I see, another candidate to fix in spark-dialect-extension